### ISSUE TITLE
Fix marked as complete issue

### ISSUE THREAD

**Issue by urdadx** [urdadx/spacely#2]
When a particular todo item is marked as completed and deleted, the next item in the list is also marked as completed although not set by the user

**Behaviour**

https://github.com/urdadx/spacely/assets/70736338/b8f8ff98-b99f-4426-af0c-0bb71fd93fc9

**Comment by Bateusz** [comment]
Hi, I resolved the initial issue you had. I noticed another problem caused by TodoList and Todo operating on separate state, which can lead to inconsistencies in saved todos. For instance, adding 3 todo and then marking a todo changes all stored todos to reflect only that single todo state. I made you appropriate PR that suits your approach.

Undesirable behavior: 
https://github.com/urdadx/spacely/assets/55596139/ea1492bb-a854-42a7-a96d-5c2687c7eb2f
